src/Comparator/BackwardsCompatibility/PropertyBased/PropertyDocumentedTypeChanged.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\ApiCompare\Comparator\BackwardsCompatibility\PropertyBased;
+
+use Roave\ApiCompare\Change;
+use Roave\ApiCompare\Changes;
+use Roave\BetterReflection\Reflection\ReflectionProperty;
+
+/**
+ * Type declarations for properties are invariant: you can't restrict the type because the consumer may
+ * write invalid values to it, and you cannot widen the type because the consumer may expect a specific
+ * type when reading.
+ */
+final class PropertyDocumentedTypeChanged implements PropertyBased
```

Comment:
Hmm, not sure about using docblocks to trigger BC breaks tbh...

![guidelines](https://user-images.githubusercontent.com/496145/38900574-c0ccfa3a-4293-11e8-81f4-ace8820256f3.jpg)

Reply:
Property docblocks are the only guarantee (although indeed really weak) that we have for now.

As I documented in #44 and #45, we still need to add them to methods too.

PHP is a shitty language, so the documented types are still the best level of "trust" that we can put in an API.

Reply:
Not convinced BC breaks based only on phpdocs (considering how incorrect they can be.........) is a good idea. If anything at the moment, I'd make these opt-in tbh, with maybe some configuration / param option.

(cue some stupid YAML-based joke that serves no purpose but to undermine my genuine concern for some kind of configurability)